rules/gcp_k8s_rules/gcp_k8s_exec_into_pod.py

```diff
@@ -1,37 +1,42 @@
 from gcp_base_helpers import get_k8s_info
 from gcp_environment import PRODUCTION_PROJECT_IDS, rule_exceptions
-from panther_base_helpers import deep_get
+from panther_base_helpers import deep_walk
 
 
 def rule(event):
     # Defaults to False (no alert) unless method is exec and principal not allowed
-    if not deep_get(
+    if not deep_walk(
         event, "protoPayload", "methodName"
-    ) == "io.k8s.core.v1.pods.exec.create" or not deep_get(event, "protoPayload", "resourceName"):
+    ) == "io.k8s.core.v1.pods.exec.create" or not deep_walk(event, "protoPayload", "resourceName"):
         return False
 
     k8s_info = get_k8s_info(event)
-    principal = k8s_info["principal"]
-    namespace = k8s_info["namespace"]
-    project_id = k8s_info["project_id"]
+    principal = deep_walk(k8s_info, "principal", default="", return_val="last")
```

Comment:
I'm open to feedback around the `return_val` usage in this Detection. It's likely that the default value can be used as well since there should only be one value returned.

Reply:
Addressed in [`61b6862` (#841)](https://github.com/panther-labs/panther-analysis/pull/841/commits/61b6862050645a74f96934a38a02d7952bf19fb3).